src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala

```diff
@@ -0,0 +1,192 @@
+package com.imaginea.activegrid.core.models
+
+/**
+  * Created by nagulmeeras on 25/10/16.
+  */
+
+import com.amazonaws.auth.AWSCredentials
+import com.amazonaws.regions.{Region, RegionUtils}
+import com.amazonaws.services.ec2.model._
+import com.amazonaws.services.ec2.{AmazonEC2, AmazonEC2Client, model}
+import com.typesafe.scalalogging.Logger
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+import scala.collection.immutable.List
+
+object AWSComputeAPI {
+  val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  def getInstances(amazonEC2: AmazonEC2 , accountInfo: AccountInfo): List[Instance] = {
+
+      val awsInstancesResult = getAWSInstances(amazonEC2)
+      val totalsecurityGroups = amazonEC2.describeSecurityGroups.getSecurityGroups.foldLeft(Map[String, SecurityGroup]())((map, sg) => map + ((sg.getGroupId, sg)))
+      val addresses = amazonEC2.describeAddresses.getAddresses.foldLeft(Map[String, Address]())((map, address) => map + ((address.getInstanceId, address)))
+      val imageIds = awsInstancesResult.foldLeft(List[String]())((list, awsInstance) => list.::(awsInstance.getImageId))
```

Comment:
for adding imageId of each AWSInstance to List.
Should i change it like this  list++List(valueToBeAdded) ?